subprojects/language-native/src/main/java/org/gradle/language/swift/internal/DefaultSwiftBinary.java

```diff
@@ -150,4 +174,73 @@ public FileCollection getObjects() {
     public Property<SwiftCompile> getCompileTask() {
         return compileTaskProperty;
     }
+
+    @Inject
+    protected NativeDependencyCache getNativeDependencyCache() {
+        throw new UnsupportedOperationException();
+    }
+
+    private class ModulePath implements MinimalFileSet, Buildable {
+        private final Configuration importPathConfig;
+
+        private Set<File> result;
+
+        ModulePath(Configuration importPathConfig) {
+            this.importPathConfig = importPathConfig;
+        }
+
+        @Override
+        public String getDisplayName() {
+            return "Module include path for " + DefaultSwiftBinary.this.toString();
+        }
+
+        @Override
+        public Set<File> getFiles() {
+            if (result == null) {
+                result = Sets.newLinkedHashSet();
+                Map<String, ModuleMap> moduleMaps = Maps.newHashMap();
+                for (ResolvedArtifactResult artifact : importPathConfig.getIncoming().getArtifacts()) {
+                    Usage usage = artifact.getVariant().getAttributes().getAttribute(Usage.USAGE_ATTRIBUTE);
+                    if (usage != null && Usage.C_PLUS_PLUS_API.equals(usage.getName())) {
+                        String moduleName;
+
+                        ComponentIdentifier id = artifact.getId().getComponentIdentifier();
+                        if (ModuleComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ModuleComponentIdentifier) id).getModule();
+                        } else if (ProjectComponentIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((ProjectComponentIdentifier) id).getProjectName();
+                        } else if (LibraryBinaryIdentifier.class.isAssignableFrom(id.getClass())) {
+                            moduleName = ((LibraryBinaryIdentifier) id).getLibraryName();
```

Comment:
I think this might only be for JVM libs?

Reply:
Yeah, I didn't realize that - I was just accounting for the known types of `ComponentModuleIdentifier`.  Removed.